### ISSUE TITLE
Exception thrown when dealing with non-semver version strings

### ISSUE THREAD

**Issue by rocher** [alire-project/alire#1676]
**Checklist**

- [x] I've included the output of `alr version`.
- [x] I've included complete steps to reproduce my issue.

**Describe the bug**
It seems that Alire cannot manage versions not following the semver rules. In particular, the external dependency `zlib1g-dev` in Ubuntu 24.04 has incompatible semver version string:

```
debug: Extracting native version from apt output: Version: 1:1.3.dfsg-3.1ubuntu2   
debug: Candidate version string: 1.3.
```

**To Reproduce**
1. `alr -vv -d search gnatcoll`

**Expected Behavior**
Simply show the `gnatcoll` crate information.

**`alr` version**
```
APPLICATION
alr version:               2.0.1
libalire version:          2.0.1
compilation date:          2024-03-21 11:06:29
compiled with version:     13.2.0

CONFIGURATION
settings folder:           /home/ada/.config/alire
cache folder:              /home/ada/.local/share/alire
vault folder:              /home/ada/.local/share/alire/releases
build folder:              /home/ada/.local/share/alire/builds
temp folder:               /run/user/1000
force flag:                FALSE
non-interactive flag:      FALSE
community index branch:    stable-1.3.0
compatible index versions: ^1.1 & <=1.3.0
indexes folder:            /home/ada/.config/alire/indexes
indexes metadata:          OK
index #1:                  (community) git+https://github.com/alire-project/alire-index#stable-1.3.0
toolchain folder:          /home/ada/.local/share/alire/toolchains
toolchain assistant:       disabled
tool #1 gnat:              gnat_native=13.2.2
tool #2 gprbuild:          gprbuild=22.0.1
system package manager:    /usr/bin/apt
distro detection disabled: FALSE

WORKSPACE
root status:               OUTSIDE
root release:              N/A
root load error:           N/A
root folder:               N/A
current folder:            /home/ada/.config/alire/indexes/community/repo

SYSTEM
distribution:              UBUNTU
host-arch:                 X86_64
os:                        LINUX
target:                    NATIVE
toolchain:                 USER
word-size:                 BITS_64
```

**`alr` logs**
Output of `alr -vv -d search gnatcoll`:
```
...
detail: Solving dependencies with options: Age order: NEWEST; Completeness: FIRST_COMPLETE; Externals: DETECT; Hinting: HINT
detail: Root dependency tree is: (gnatcoll~24.0.0 and zlib*)
detail: No user pins to apply                  
debug: Not reloading crate gnatcoll            
debug: Not reloading crate gnatcoll            
debug: Skipping detection for crate without externals: gnatcoll
debug: Not reloading crate gnatcoll            
debug: Not reloading crate gnatcoll            
debug: Not reloading crate gnatcoll            
debug: Skipping detection for crate without externals: gnatcoll
debug: Not reloading crate gnatcoll            
debug: Not reloading crate zlib                
debug: Not reloading crate zlib                
debug: Looking for externals for crate: zlib   
debug: Detecting via provider: zlib            
debug: Status start: Looking for external crate: zlib
debug: Attempting detection of available external: zlib                            
debug: Looking for system packages that provide crate: zlib                        
detail: Looking for system package: zlib1g-dev                                     
debug: Selected name for tempfile: /run/user/1000/alr-rmcj.tmp when at dir: /home/ada/.config/alire/indexes/community/repo
detail: Spawning: ["apt-cache", "-q", "show", "zlib1g-dev"] > /run/user/1000/alr-rmcj.tmp
debug: Deleting temporary file /run/user/1000/alr-rmcj.tmp...                      
debug: Command exited with valid code: 0                                           
debug: Extracting native version from apt output: Version: 1:1.3.dfsg-3.1ubuntu2   
debug: Candidate version string: 1.3.                                              
debug: Status ended: Looking for external crate: zlib                              
debug: Status ended: Solving dependencies                                          
debug: Status ended: gn                        
debug: Status ended: Searching
debug: ---8<--- Exception dump begin ---8<---
debug: SEMANTIC_VERSIONING.MALFORMED_INPUT
debug: Empty point number
debug: raised SEMANTIC_VERSIONING.MALFORMED_INPUT : Empty point number
[/usr/local/bin/alr]
0x457d88 Semantic_Versioning.Parse.Eat_Number at semantic_versioning.adb:78
0x45981b Semantic_Versioning.Parse.Accept_Number at semantic_versioning.adb:142
0x459b07 Semantic_Versioning.Parse.Accept_Number at semantic_versioning.adb:159
0x459b07 Semantic_Versioning.Parse.Accept_Number at semantic_versioning.adb:159
0x45af3e Semantic_Versioning.Parse at semantic_versioning.adb:179
0x9c0403 Alire.Origins.Deployers.System.Apt.Detect at alire-origins-deployers-system-apt.adb:73
0x89b0b9 Alire.Externals.From_System.Detect at alire-externals-from_system.adb:46
0x8b1a18 Alire.Externals.Lists.Detect at alire-externals-lists.adb:34
0x926d0d Alire.Index.Detect_Externals at alire-index.adb:189
0xd111c2 Alire.Solver.Resolve.Detect_Unavailable_Direct_Dependencies.2 at alire-solver.adb:1220
0xcf76b7 Alire.Solver.Resolve at alire-solver.adb:1296
0xcf4f06 Alire.Solver.Is_Resolvable at alire-solver.adb:182
0x4cc6d7 Alr.Commands.Search.Execute at alr-commands-search.adb:54
0x4ce5ef Alr.Commands.Search.Execute at alr-commands-search.adb:111
0x4cfbb0 Alr.Commands.Search.Execute at alr-commands-search.adb:225
0x4d0f73 Alr.Commands.Search.Execute at alr-commands-search.adb:273
0x4cb8b8 Alr.Commands.Search.Execute at alr-commands-search.adb:302
0x4cbce7 Alr.Commands.Search.Execute at alr-commands-search.adb:23
0x53939b Alr.Commands.Sub_Cmd.Execute at clic-subcommand-instance.adb:749
0x54587e Alr.Commands.Execute at alr-commands.adb:561
0x436183 Alr.Main at alr-main.adb:13
0x438de1 Main at b__alr-main.adb:2257
[/lib/x86_64-linux-gnu/libc.so.6]
0x7adc7622a1c8
0x7adc7622a289
[/usr/local/bin/alr]
0x43601c _start at ???
0xfffffffffffffffe

debug: --->8--- Exception dump end ----->8---
stderr: SEMANTIC_VERSIONING.MALFORMED_INPUT
stderr: Empty point number
stderr: raised SEMANTIC_VERSIONING.MALFORMED_INPUT : Empty point number
[/usr/local/bin/alr]
0x457d88 Semantic_Versioning.Parse.Eat_Number at semantic_versioning.adb:78
0x45981b Semantic_Versioning.Parse.Accept_Number at semantic_versioning.adb:142
0x459b07 Semantic_Versioning.Parse.Accept_Number at semantic_versioning.adb:159
0x459b07 Semantic_Versioning.Parse.Accept_Number at semantic_versioning.adb:159
0x45af3e Semantic_Versioning.Parse at semantic_versioning.adb:179
0x9c0403 Alire.Origins.Deployers.System.Apt.Detect at alire-origins-deployers-system-apt.adb:73
0x89b0b9 Alire.Externals.From_System.Detect at alire-externals-from_system.adb:46
0x8b1a18 Alire.Externals.Lists.Detect at alire-externals-lists.adb:34
0x926d0d Alire.Index.Detect_Externals at alire-index.adb:189
0xd111c2 Alire.Solver.Resolve.Detect_Unavailable_Direct_Dependencies.2 at alire-solver.adb:1220
0xcf76b7 Alire.Solver.Resolve at alire-solver.adb:1296
0xcf4f06 Alire.Solver.Is_Resolvable at alire-solver.adb:182
0x4cc6d7 Alr.Commands.Search.Execute at alr-commands-search.adb:54
0x4ce5ef Alr.Commands.Search.Execute at alr-commands-search.adb:111
0x4cfbb0 Alr.Commands.Search.Execute at alr-commands-search.adb:225
0x4d0f73 Alr.Commands.Search.Execute at alr-commands-search.adb:273
0x4cb8b8 Alr.Commands.Search.Execute at alr-commands-search.adb:302
0x4cbce7 Alr.Commands.Search.Execute at alr-commands-search.adb:23
0x53939b Alr.Commands.Sub_Cmd.Execute at clic-subcommand-instance.adb:749
0x54587e Alr.Commands.Execute at alr-commands.adb:561
0x436183 Alr.Main at alr-main.adb:13
0x438de1 Main at b__alr-main.adb:2257
[/lib/x86_64-linux-gnu/libc.so.6]
0x7adc7622a1c8
0x7adc7622a289
[/usr/local/bin/alr]
0x43601c _start at ???
0xfffffffffffffffe
```

**Comment by mosteo** [comment]
Thanks for the report, I see the root of the problem is that in `22.04` there is a point number (`1:1.2.11.dfsg-2ubuntu9.2`) that is no longer there in `24.04`. This is likely confusing even the permissive parser in our `semver` implementation.

**Comment by mosteo** [comment]
Fixed in #1704.
